src/functions/aggregate.js

```diff
@@ -5,30 +5,30 @@ export const aggregate = array => {
   // First key is data series for Highcharts of aggregated impressions.
   // Second key is data series for Highcharts of aggregated clicks.
   const groupByDate = _.groupBy(array, "Date");
-  const uniqueDates = Object.keys(groupByDate);
-  const impressions = uniqueDates.map(item => {
-    const sumImp = groupByDate[item].reduce((acc, obj) => {
-      return acc + obj.Impressions;
-    }, 0);
-    return [parseInt(item), sumImp];
-  });
-  const clicks = uniqueDates.map(item => {
-    const sumClick = groupByDate[item].reduce((acc, obj) => {
-      return acc + obj.Clicks;
-    }, 0);
-    return [parseInt(item), sumClick];
-  });
   return {
     impressions: {
       type: "line",
       name: "Impressions",
       yAxis: 1,
-      data: impressions
+      data: Object.keys(groupByDate).map(item => {
+        const sumImpressions = groupByDate[item].reduce(
+          (accumulator, object) => {
+            return accumulator + object.Impressions;
+          },
+          0
+        );
+        return [parseInt(item), sumImpressions];
+      })
     },
     clicks: {
       type: "line",
       name: "Clicks",
-      data: clicks
+      data: Object.keys(groupByDate).map(item => {
```

Comment:
basically the same code as above (for impressions). extract to a function and add tests

Reply:
Function extracted and tests added